### DISCUSSION TITLE
Self-hosting module server

### DISCUSSION THREAD

**Discussion by reggi** [denoland/deno#15550] [Ideas]
I'm curious if any projects exist that would allow me to configure a small server ideally like https://github.com/denoland/deno_blog that allow me to spin up a host for deno modules. In the simplest incarnation being just with an array of repos and deno runs sh / clones them to the server and serves the pages with the correct mime types. Perhaps tagged version support.

Would be ideal to have github webhook support like third-party modules on deno.land do!

This would be a parallel to npm's https://verdaccio.org/

Something that could possibly have enterprise interest to allow companies with a VPN to host their own code within their network 👀

**Comment by justinmchase** [comment]
I actually made one of these for my company, it is more like a proxy and cache for our various internal sources of packages such as github enterprise and artifactory. I plan on bringing that forward as OSS as we adopt deno more.

**Reply by crackedbeefcake** [reply]
did anything ever come of this? 
i am looking to setup and run a registry soon and, as i've never done this, any experience and insight would be very welcome. 
best

**Reply by justinmchase** [reply]
I think what I had is now pretty obsolete. What you'd probably want is to self-host a JSR registry.

Here is an interesting comment but the deno team on the matter!

https://github.com/jsr-io/jsr/issues/203#issuecomment-1992050655
